packages/mask/src/extension/popups/modals/ConnectProvider/index.tsx

```diff
@@ -0,0 +1,182 @@
+import { PopupModalRoutes, type NetworkPluginID, PopupRoutes } from '@masknet/shared-base'
+import { Web3 } from '@masknet/web3-providers'
+import { timeout } from '@masknet/kit'
+import urlcat from 'urlcat'
+
+interface StyleProps {
+    loading: boolean
+    timeout: boolean
+}
+
+const useStyles = makeStyles<StyleProps>()((theme, { loading, timeout }) => ({
+    tips: {
+        fontSize: 14,
+        color: theme.palette.maskColor.third,
+        fontWeight: 700,
+        lineHeight: '18px',
+        textAlign: 'center',
+    },
+    icon: {
+        '@keyframes spinner': {
+            to: {
+                transform: 'rotate(360deg)',
+            },
+        },
+        position: 'relative',
+        width: 48,
+        height: 48,
+        display: 'flex',
+        justifyContent: 'center',
+        alignItems: 'center',
+        ...(loading
+            ? {
+                  '&:before': {
+                      content: "''",
+                      boxSizing: 'border-box',
+                      position: 'absolute',
+                      top: -5,
+                      left: -5,
+                      width: 56,
+                      height: 56,
+                      borderRadius: '50%',
+                      border: `2px solid ${theme.palette.maskColor.main}`,
+                      borderTopColor: theme.palette.maskColor.second,
+                      animation: 'spinner 2s linear infinite',
+                  },
+              }
+            : {}),
+
+        ...(timeout
+            ? {
+                  border: `2px solid ${theme.palette.maskColor.danger}`,
+                  borderRadius: '50%',
+                  padding: 10,
+              }
+            : {}),
+    },
+}))
+
+export const ConnectProviderModal = memo<ActionModalBaseProps>(function ConnectProviderModal({ ...rest }) {
+    const { t } = useI18N()
+    const navigate = useNavigate()
+    const modalNavigate = useModalNavigate()
+    const { closeModal } = useActionModal()
+    const location = useLocation()
+    const { pluginID } = useNetworkContext<NetworkPluginID.PLUGIN_EVM>()
+
+    const { showSnackbar } = usePopupCustomSnackbar()
+
+    const { Provider } = useWeb3State<void, NetworkPluginID.PLUGIN_EVM>(pluginID)
+    const { providerType } = useMemo(() => {
+        const params = new URLSearchParams(location.search)
+        const providerType = params.get('providerType')
+
+        return {
+            providerType: providerType ? providerType : undefined,
+        }
+    }, [])
+
+    const provider = useProviderDescriptor<void, NetworkPluginID.PLUGIN_EVM>(pluginID, providerType)
+
+    const providerExist = useMemo(() => {
+        return Provider?.isReady(provider.type)
+    }, [providerType, Provider])
+
+    const handleClose = useCallback(() => {
+        navigate(urlcat(PopupRoutes.Personas, { disableNewWindow: true }), { replace: true })
+    }, [])
+
+    const { loading, error, retry } = useAsyncRetry(async () => {
```

Comment:
Attention: It will try to connect a provider if the component gets mounted. Use a callback function will be a better choice.